### PR TITLE
Istio-agent should cache root cert read from file

### PR DESCRIPTION
**Pull request by JimmyCYJ**
https://github.com/istio/istio/issues/20951

### CODE REVIEW

File: security/pkg/nodeagent/cache/secretcache.go [hunk 1]
```diff
@@ -245,7 +254,8 @@ func (sc *SecretCache) GenerateSecret(ctx context.Context, connectionID, resourc
 			return nil, err
 		}
 		// This is not stored - envoy will refresh when the cert is about to expire.
-		cacheLog.Infoa("GenerateSecret from file", resourceName)
+		cacheLog.Infoa("GenerateSecret from file ", resourceName)
+		sc.secrets.Store(connKey, *ns)
```
Comment (howardjohn) [inline security/pkg/nodeagent/cache/secretcache.go]: nit - the comment says it is not stored but we store it

Comment (JimmyCYJ) [reply security/pkg/nodeagent/cache/secretcache.go]: Nice catch! Thanks! This comment is not correct, I have updated it.

**Comment by howardjohn** [comment]
I confirmed this fixes the problem

File: security/pkg/nodeagent/cache/secretcache.go [hunk 2]
```diff
@@ -244,8 +253,9 @@ func (sc *SecretCache) GenerateSecret(ctx context.Context, connectionID, resourc
 				conIDresourceNamePrefix, err)
 			return nil, err
 		}
-		// This is not stored - envoy will refresh when the cert is about to expire.
-		cacheLog.Infoa("GenerateSecret from file", resourceName)
+		// TODO(JimmyCYJ): need a file watcher to detect file updates and push new secret to clients.
```
Comment (myidpt) [inline security/pkg/nodeagent/cache/secretcache.go]: @lei-tang To confirm: we need a file watcher here, right?

File: security/pkg/nodeagent/cache/secretcache.go [hunk 3]
```diff
@@ -74,13 +83,13 @@ const (
 	notifyK8sSecretTimeout = 30 * time.Second
 
 	// The well-known path for an existing certificate chain file
-	existingCertChainFile = "./etc/certs/cert-chain.pem"
+	CertChainFilePath = "./etc/certs/cert-chain.pem"
```
Comment (myidpt) [inline security/pkg/nodeagent/cache/secretcache.go]: Start with lowercases, and name as "defaultCertChainFilePath"? Same for the other two.

File: security/pkg/nodeagent/cache/secretcache.go (hunk 2 above)
Comment (JimmyCYJ) [reply security/pkg/nodeagent/cache/secretcache.go]: There is a TODO at line 237 in this file https://github.com/istio/istio/blob/32087603be189c49dcc39f94c46554f0adc4e98c/security/pkg/nodeagent/cache/secretcache.go#L228

File: security/pkg/nodeagent/cache/secretcache.go (hunk 3 above)
Comment (JimmyCYJ) [reply security/pkg/nodeagent/cache/secretcache.go]: Done, thanks.

File: security/pkg/nodeagent/cache/secretcache.go (hunk 2 above)
Comment (myidpt) [reply security/pkg/nodeagent/cache/secretcache.go]: Thanks.

**Pull request merged**